### PR TITLE
Shell script for updating shapes and api files

### PR DESCRIPTION
**Pull request by adam-fowler**
- Gets models from aws-sdk-go
- Copies them into aws-sdk-swift project
- Builds code generator
- Runs code generator
- Builds service files

Thought we could update the model files as the last build was done 9 months ago. Then thought automating this would make it a lot quicker and easier to keep these up to date.

### CODE REVIEW

**Comment by adam-fowler** [comment]
Added options to 
- specify a aws-sdk-go version number 
- commit changes immediately

File: scripts/update_models.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+
+set -e
```
Comment (Yasumoto) [inline scripts/update_models.sh]: How about `set -eux` ? 🙏

File: scripts/update_models.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+usage()
+{
+  echo "Usage: update_models.sh -c [ -v MODELS_VERSION_NUMBER ]"
+  exit 2
+}
+
+get_aws_sdk_go()
+{
+  # clone aws-sdk-go into folder
```
Comment (Yasumoto) [inline scripts/update_models.sh]: Small nit, any chance you could do something like

```
DESTINATION_FOLDER="$1"
BRANCH_NAME="$2"
```

here and elsewhere to make it clear what the positional args are for? 🙏

File: scripts/update_models.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    swift build
+}
+
+check_for_local_changes()
+{
+    LOCAL_CHANGES=`git status --porcelain`
+    if [ -n "$LOCAL_CHANGES" ] && [ "$COMMIT_CHANGES" == 1 ]; then
+        echo "You have local changes already and you have requested to commit changes after this script has run.\nEither remove the -c option or stash your local changes."
+        usage
+        exit
```
Comment (Yasumoto) [inline scripts/update_models.sh]: `usage` should already call `exit 2` so this isn't necessary

File: scripts/update_models.sh [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    fi
+}
+
+commit_changes()
+{
+    COMMIT_MSG="Sync models with aws-sdk-go "$AWS_MODELS_VERSION
```
Comment (Yasumoto) [inline scripts/update_models.sh]: I almost always double-quote my variables in shell, so this should be:

`COMMIT_MSG="Sync models with aws-sdk-go $AWS_MODELS_VERSION"`

File: scripts/update_models.sh [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        exit ;;
+    esac
+done
+
+
+trap cleanup EXIT
+
+check_for_local_changes
+
+#create temp folder
```
Comment (Yasumoto) [inline scripts/update_models.sh]: I'd suggest removing these, your echo statements do a great job of describing what's going on already.

**Comment by adam-fowler** [comment]
Updated the script with your change requests, ran shellcheck on it and fixed issues.
I'm happy for shellcheck to run on all shell scripts if you want to add it in

**Comment by adam-fowler** [comment]
@jonnymacs @Yasumoto Are you guys happy for me to do a rebuild of all the Shape files?

**Pull request merged**